### DISCUSSION TITLE
Can I put link in stencil?

### DISCUSSION THREAD

**Discussion by dwanl** [clientIO/joint#1659] [Q&A] [answered]

**Comment by kumilingus** [comment, marked as answer]
I am really sorry for a massive delay. It's not supported out-of-the-box at this moment.

Here's a workaround though.

Create an element that looks like a link.
```js
const LinkElement = joint.dia.Element.define('example.LinkElement', {
    attrs: {
        bg: {
            refWidth: '100%',
            refHeight: '100%',
            fill: 'transparent'
        },
        line: {
            refD: 'M 0 1 1 0',
            stroke: '#333',
            strokeWidth: 2,
            targetMarker: {
                'type': 'path',
                'd': 'M 10 -5 0 0 10 5 z'
            }
        }
    }
}, {
    markup: [{
        tagName: 'rect',
        selector: 'bg',
    }, {
        tagName: 'path',
        selector: 'line'
    }]
});
```

Add `LinkElement` to the stencil.
```js
stencil.load([new LinkElement()]);
```

Replace the element with a link on `element:dragend`.
```js
stencil.on('element:dragend', function(cloneView, _evt, cloneArea, validDropTarget) {
    if (!validDropTarget) return;
    if (cloneView.model.get('type') === 'example.LinkElement') {
        // do not add the stencil clone to the graph
        stencil.cancelDrag({ dropAnimation: false });
        // add an actual link instead
        const link = new joint.shapes.standard.Link();
        link.source(cloneArea.bottomLeft());
        link.target(cloneArea.topRight());
        link.addTo(graph);
    }
});
```

Here's an example:
https://codepen.io/jointjs/pen/qBPzMbq

**Reply by naresh-FD** [reply]
Thanks, @kumilingus but what we need is to bend as well add link to the elements with the same link similar to the mix bus Example. like a cad drawing. 

Attached a sample image for reference.

![image](https://user-images.githubusercontent.com/50324374/204757614-33387246-d7bc-4720-a18c-6be5942585fe.png)

![image](https://user-images.githubusercontent.com/50324374/204757303-937340e0-9b76-4cc0-b448-bc79e77c49e4.png)

**Reply by kumilingus** [reply]
Could you please elaborate? Does your question have anything to do with the stencil? Do you want to show bended links in the stencil or bend links after they were dropped to the paper?

**Reply by naresh-FD** [reply]
Hi @kumilingus

The exact requirement is 

1. we need to drag and drop a line. So that we can achieve a pipeline kind of structure similar [AutoCAD ](https://user-images.githubusercontent.com/50324374/204757614-33387246-d7bc-4720-a18c-6be5942585fe.png) image that we have attached above.

2. Hope you can see that the Autocad image has a curved bend as well as a sharp bend. we need to know how to achieve the above with joint js. with a link or else by line

**Reply by kumilingus** [reply]
Please create a separate discussion for this.
